arm-software/embedded/arm-runtimes/test-support/run_qemu.py

```diff
@@ -44,6 +45,13 @@ def run_qemu(
     else:
         qemu_params += ["-device", f"loader,file={image},cpu-num=0"]
 
+    # Enable tracing: disassembly, CPU state, interrupts and guest errors like
+    # invalid instructions. One instruction at a time.
+    if trace:
+        qemu_params += ["-singlestep"]
```

Comment:
`-singlestep` is [deprecated](https://www.qemu.org/docs/master/about/removed-features.html#singlestep-removed-in-9-0).

Reply:
I think the singlestep feature is very important for debugging. I understand that detecting the version can feel awkward, but I'd still do it.

Reply:
I gave it a try, please have a look, thanks!